### PR TITLE
[flutter_tools] split features into 2 libraries, migrate to null safety

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards #71511

FlutterFeatures was already separate from Features, just move to a separate library to make it easier to migrate dependencies.

### CODE REVIEW

File: packages/flutter_tools/lib/src/features.dart [hunk 1]
```diff
@@ -2,19 +2,12 @@
...
-import 'package:meta/meta.dart';
-
-import 'base/config.dart';
 import 'base/context.dart';
-import 'base/platform.dart';
-import 'version.dart';
 
 /// The current [FeatureFlags] implementation.
 ///
 /// If not injected, a default implementation is provided.
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/features.dart:9]: > If not injected, a default implementation is provided.

That's not true, right? I think you may hit a null error in context_test:

https://source.corp.google.com/piper///depot/google3/mobile/flutter/cli/lib/context.dart;l=673?q=%20FeatureFlags&ss=piper%2FGoogle%2FPiper:google3%2Fmobile%2Fflutter%2Fcli%2F

Similar to the change I made in:
https://critique-ng.corp.google.com/cl/365154959/depot/google3/mobile/flutter/cli/test/context_test.dart

You may need to add this to `context_test` `Honors a specified project platform`:
```dart
FeatureFlags: () => Google3Features(),
```
And make `Google3Features` visible for testing.

**Review by jmagman — APPROVED** [review]
If the Google Tests pass, LGTM.  Approving to kick that off.

File: packages/flutter_tools/lib/src/features.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/features.dart:9]: Ahh yeah, I think this used to have a default...

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/features.dart:9]: you were correct :D

Comment (jmagman) [reply packages/flutter_tools/lib/src/features.dart:9]: 🔮

**Pull request merged**